lib/spack/spack/directory_layout.py

```diff
@@ -262,7 +262,19 @@ def create_install_directory(self, spec):
         if prefix:
             raise InstallDirectoryAlreadyExistsError(prefix)
 
-        mkdirp(self.metadata_path(spec))
+        # Create install directory with properly configured permissions
+        # Cannot import at top of file
+        from spack.package_prefs import get_package_dir_permissions
+        from spack.package_prefs import get_package_group
```

Comment:
Why is this imported here?  Import `spack.package_prefs` at the top level and just use these explicitly.